src/main/scala/org/graphframes/lib/ConnectedComponents.scala

```diff
@@ -356,12 +362,14 @@ object ConnectedComponents extends Logging {
 
       ee.persist(intermediateStorageLevel)
 
-      // test convergence
+      minNbrs1 = minNbrs(ee) // src >= min_nbr
+        .persist(intermediateStorageLevel)
 
+      // test convergence
       // Taking the sum in DecimalType to preserve precision.
       // We use 20 digits for long values and Spark SQL will add 10 digits for the sum.
       // It should be able to handle 200 billion edges without overflow.
-      val (currSum, cnt) = ee.select(sum(col(SRC).cast(DecimalType(20, 0))), count("*")).rdd
+      val (currSum, cnt) = minNbrs1.select(sum(col(MIN_NBR).cast(DecimalType(20, 0))), count("*")).rdd
```

Comment:
This is the convergence critieria correction.

Ref: Lemma 8 in paper https://dl.acm.org/doi/pdf/10.1145/2670979.2670997